src/FakeItEasy/Expressions/CallConstraintDescriber.cs

```diff
@@ -15,15 +16,23 @@ internal static class CallConstraintDescriber
         /// matcher to the supplied writer.
         /// </summary>
         /// <param name="writer">The writer on which to describe the call.</param>
+        /// <param name="callTarget">The object on which the method was called.</param>
         /// <param name="method">The method to describe.</param>
         /// <param name="argumentConstraints">The argument constraints applied to the method.</param>
-        public static void DescribeCallOn(IOutputWriter writer, MethodInfo method, IEnumerable<IArgumentConstraint> argumentConstraints)
+        public static void DescribeCallOn(IOutputWriter writer, object? callTarget, MethodInfo method, IEnumerable<IArgumentConstraint> argumentConstraints)
         {
-            writer.Write(method.DeclaringType!);
-            writer.Write(".");
+            WriteObjectType(writer, method.DeclaringType);

```

Comment:
Please restore the "!". It's causing a [warning CS8604 in the build](https://ci.appveyor.com/project/FakeItEasy/fakeiteasy/builds/42130140/job/0xgev6ijlyhomatl#L67).